core/src/main/java/hudson/model/Slave.java

```diff
@@ -383,6 +385,8 @@ public URL getURL() throws MalformedURLException {
             
             if (name.equals("hudson-cli.jar"))  {
                 name="jenkins-cli.jar";
+            } else if (name.equals("slave.jar") || name.equals("remoting.jar")) {
+                name = "lib/" + Which.jarFile(Channel.class).getName();
```

Comment:
I'm not sure this change is compatible since one may patch WAR file and put different implementations